### PR TITLE
Chuck's changes

### PR DESCRIPTION
**Pull request by fvitt**
new file:   fixcorecol.F90
        modified:   carmastate_mod.F90
        modified:   coremasscheck.F90
        modified:   fixcorecol.F90
        modified:   microfast.F90
        modified:   newstate.F90
        modified:   newstate_calc.F90
        modified:   vertical.F90

### CODE REVIEW

File: fixcorecol.F90 [hunk 1]
```diff
@@ -0,0 +1,119 @@
+! Include shortname defintions, so that the F77 code does not have to be modified to
+! reference the CARMA structure.
+#include "carma_globaer.h"
+
+!! This routine checks the column for errors where the sum of the core masses
+!! is larger than the concentration element. This implies a negative mass
+!! of the concentration element, which is not physical.
+!!
+!! This routine attempts to conserving mass by using mass from the positive
```
Comment (cbardeen) [inline fixcorecol.F90]: conserve not conserving

File: vertical.F90 [hunk 2]
```diff
@@ -89,22 +99,44 @@ subroutine vertical(carma, cstate, rc)
...
 
+
+  ! Advection can cause errors in tracer/tracer relationship that can cause
+  ! negative values for the concentration element. Use a mass conserving
+  ! fixer to make sure there are no negative values.
+  call fixcorecol(carma, cstate, rc)
+  if (rc < RC_OK) return
+
+
+  ! Now the the column has been fixed up, look to see how much has mass
```
Comment (cbardeen) [inline vertical.F90]: much has mass -> much mass

**Review by cbardeen — APPROVED** [review]

File: fixcorecol.F90 (hunk 1 above)
Comment (fvitt) [reply fixcorecol.F90]: fixed

File: vertical.F90 (hunk 2 above)
Comment (fvitt) [reply vertical.F90]: fixed

**Pull request merged**